src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -120,20 +121,37 @@ private BoundExpression MakeUtf8Span(BoundExpression node, IReadOnlyList<byte>?
             return result;
         }
 
-        private byte[]? GetUtf8ByteRepresentation(BoundUtf8String node)
+        internal static bool TryGetUtf8ByteRepresentation(
+            string s,
+            [NotNullWhen(returnValue: true)] out byte[]? result,
+            [NotNullWhen(returnValue: false)] out string? error)
         {
-            var utf8 = new System.Text.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
```

Comment:
>var utf8 = new System.Text.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true); [](http://example.com/codeflow?start=12&length=107)

Is there a specific reason behind caching this value? Would it make sense to initialize the cache on demand rather than using a field initializer?

Reply:
> Is there a specific reason behind caching this value?

No specific reason other than avoiding unnecessary allocations.

> Would it make sense to initialize the cache on demand rather than using a field initializer?

Yes, thanks.